perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ResourcesManagerBlImpl.java

```diff
@@ -51,6 +51,44 @@ public Resource createResource(PerunSession sess, Resource resource, Vo vo, Faci
 		return resource;
 	}
 
+	public Resource copyResource(PerunSession sess, Resource resource, String newResourceName, Vo destinationVo, Facility destinationFacility) throws ResourceExistsException, InternalErrorException {
+		//is the new resource name not taken yet?
+		Resource newResource = new Resource();
+		try{
+			Resource existingResource = getResourcesManagerImpl().getResourceByName(sess, destinationVo, destinationFacility, newResourceName);
+			throw new ResourceExistsException(existingResource);
+		} catch (ResourceNotExistsException e){
+			//if it is not taken we create new Resource in DB, then copy everything possible from template resource to our new one.
+			newResource.setName(newResourceName);
+			newResource = getResourcesManagerImpl().createResource(sess, destinationVo, newResource, destinationFacility);
+
+			// We don't copy group_resource/member_resource attributes because they are different for each VO and needs to be set manually.
+			List<Attribute> templateResourceAttributes = perunBl.getAttributesManagerBl().getAttributes(sess,resource);
+			for (Attribute resourceAttribute : templateResourceAttributes) {
+				try {
+					if (!resourceAttribute.getNamespace().startsWith(AttributesManager.NS_RESOURCE_ATTR_VIRT)) {
```

Comment:
I'm not sure if i can copy virtual resource attributes, if yes then i will remove this if.

Reply:
You can't copy virtual attributes. They are computed from other values in Perun. You can copy just attributes with set value (def and opt).